cartography/intel/rapid7/util.py

```diff
@@ -0,0 +1,303 @@
+
+    logger.warning("extract_rapid7_configurations_subscriptionid: %s", subscription_id)
+    return subscription_id
+
+
+def extract_rapid7_configurations_resourcegroup(configurations):
+    """
+    Extract from json, specific keyvalue pair: azure resource group
+
+    Input: dataframe row configurations column
+    """
+
+    if configurations is None or not isinstance(configurations, list):
+        return None
+
+    rg = None
+    for line in configurations:
+        if line["name"] == "azure":
+            logger.warning("line: %s", line)
+            if isinstance(line["value"], str):
+                azure = json.loads(line["value"])
+            else:
+                azure = line["value"]
+            rg = extract_rg_from_resourceid(azure["resourceId"])
+
+    logger.warning("extract_rapid7_configurations_resourcegroup: %s", rg)
+    return rg
+
+
+def extract_rapid7_configurations_resourceid(configurations):
+    """
+    Extract from json, specific keyvalue pair: azure resource id
+
+    Input: dataframe row configurations column
+    """
+
+    if configurations is None or not isinstance(configurations, list):
+        return None
+
+    resource_id = None
+    for line in configurations:
+        if line["name"] == "azure":
+            logger.warning("line: %s", line)
+            if isinstance(line["value"], str):
+                azure = json.loads(line["value"])
+            else:
+                azure = line["value"]
+            resource_id = azure["resourceId"]
+
+    logger.warning("extract_rapid7_configurations_resourceid: %s", resource_id)
+    return resource_id
+
+
+def extract_rapid7_configurations_instanceid(configurations):
+    """
+    Extract from json, specific keyvalue pair: azure instance id
+
+    Input: dataframe row configurations column
+    """
+
+    if configurations is None or not isinstance(configurations, list):
+        return None
+
+    instance_id = None
+    for line in configurations:
+        if line["name"] == "azure":
+            logger.warning("line: %s", line)
+            if isinstance(line["value"], str):
+                azure = json.loads(line["value"])
+            else:
+                azure = line["value"]
+            instance_id = azure["instanceId"]
+
+    logger.warning("extract_rapid7_configurations_instanceid: %s", instance_id)
+    return instance_id
+
+
+def extract_rapid7_history_lastseen(history):
+    """
+    Extract from rapid7 history, last seen/scan time
+    """
+    lastseen = history[0]["date"]
+    for entry in history:
+        if lastseen < entry["date"]:
+            lastseen = entry["date"]
+    return lastseen
+
+
+def extract_rapid7_history_firstseen(history):
+    """
+    Extract from rapid7 history, first seen/scan time
+    """
+    firstseen = history[0]["date"]
+    for entry in history:
+        if firstseen > entry["date"]:
+            firstseen = entry["date"]
+    return firstseen
+
+
+# pylint: disable=too-many-arguments,too-many-locals
```

Comment:
is this necessary?

Reply:
As said above, I reviewed files with pylint and black.
This avoid triggering pylint errors.

Reply:
is this fine or should I remove those?

Reply:
Try to address the pylint issues.

Reply:
done